scripts/start_qp_spades

```diff
@@ -19,7 +44,58 @@ from qp_spades import plugin
 @click.argument('output_dir', required=True)
 def execute(url, job_id, output_dir):
     """Executes the task given by job_id and stores the output in output_dir"""
-    plugin(url, job_id, output_dir)
+    # there are basically 2 different kinds of jobs: register (commands)
+    # and everything else
+    if 'register' in job_id:
+        plugin(url, job_id, output_dir)
+    else:
+        qclient = _client_connect(url)
+        job_info = qclient.get_job_info(job_id)
+        parameters = job_info['parameters']
+
+        qclient.update_job_step(
+            job_id, "Step 1 of 4: Collecting info and generating submission")
+        artifact_id = parameters['input']
+        del parameters['input']
+
+        # Get the artifact filepath information
+        artifact_info = qclient.get("/qiita_db/artifacts/%s/" % artifact_id)
+        # artifact_info['files'] returns a dict of {filetype: files} so we
+        # need to parse the actual input directory and make sure it's the same
+        # for all the files; it should _never_ be different but better to have
+        # a test for this
+        directory = {dirname(ffs) for _, fs in artifact_info['files'].items()
+                     for ffs in fs}
+        if len(directory) != 1:
+            msg = ('There is more than one input directory [%s]; please '
+                   'contact qiita.help@gmail.com' % str(directory))
+            return False, None, msg
+        else:
+            directory = directory.pop()
+
+        # Get the artifact metadata
+        prep_info = qclient.get('/qiita_db/prep_template/%s/'
+                                % artifact_info['prep_information'][0])
+        df = pd.read_csv(prep_info['prep-file'], sep='\t', dtype='str',
+                         na_values=[], keep_default_na=True)
+        if 'run_prefix' not in df.columns:
+            return False, None, 'Missing run_prefix column in your preparation'
```

Comment:
Would it be feasible for this to be ensured by Qiita at prep creation?

Reply:
I guess, you mean that this becomes a required field for Qiita to create the prep template, correct? AFAIK it's required to link the raw artifact to the prep but not to create it. Thus, in theory, it's a requirement but it's also possible for users to update the prep after creation and modify this field. In other words, this might not be necessary but it also might prevent problems.

Reply:
Okay. It may be beneficial at a future time (ie out-of-scope here) to consider whether workflows/plugins should have a formalized way to state requirements about prep/sample information?